crates/turbopack-core/src/introspect/asset.rs

```diff
@@ -33,6 +37,31 @@ fn reference_ty() -> StringVc {
     StringVc::cell("reference".to_string())
 }
 
+#[turbo_tasks::function]
+fn placed_or_parallel_reference_ty() -> StringVc {
+    StringVc::cell("placed/parallel reference".to_string())
+}
+
+#[turbo_tasks::function]
+fn placed_reference_ty() -> StringVc {
+    StringVc::cell("placed reference".to_string())
+}
+
+#[turbo_tasks::function]
+fn parallel_reference_ty() -> StringVc {
+    StringVc::cell("parallel reference".to_string())
+}
+
+#[turbo_tasks::function]
+fn separate_reference_ty() -> StringVc {
+    StringVc::cell("separate reference".to_string())
+}
+
+#[turbo_tasks::function]
+fn async_reference_ty() -> StringVc {
+    StringVc::cell("async reference".to_string())
+}
```

Comment:
Would be cool if we had static-like Vcs.

```rust
turbo_tasks::static! {
    static PARALLEL_REFERENCE_TY: StringVc = StringVc::cell("parallel reference".to_string());
}
```

This is essentially the same as a tt::function so probably not worth it.